projects/sandbox/train/train.py

```diff
@@ -115,6 +118,18 @@ def main(
         glitch_frac,
         device,
     )
+
+    # TODO: hard-coding num_ifos into preprocessor. Should
+    # we just expose this as an arg? How will this fit in
+    # to the broader-generalization scheme?
+    preprocessor = WhiteningTransform(2, sample_rate, kernel_length)
+
+    # TODO: make this a `train_dataset.background` `@property`?
+    background = torch.stack(
+        [train_dataset.hanford_background, train_dataset.livingston_background]
+    )
+    preprocessor.fit(background)
+
```

Comment:
Add preprocessor type as argument so we can use StandardScaler?

Reply:
So I was thinking about something like this, but preprocessors will have their own keyword arguments (e.g. `StandardScaler` doesn't have any args besides `num_ifos`). This might invite a wrapper like `architecturize`, but this starts to get complicated pretty fast and since in principle the preprocessing method shouldn't change with anywhere near the frequency of the architecture (there should be a "right" way to do preprocessing for a particular use case), I don't think it's too much to ask individual projects to build their preprocessing methods explicitly